### PR TITLE
 Adapt to latest version of `wifi-x-nucleo-idw01m1`

### PR DESCRIPTION
**Pull request by betzw**

### CODE REVIEW

**Comment by JanneKiiskila** [comment]
@betzw - ready yet? The hash is a bit old for the actual driver.
Plus I'd want to fix the AT-parser hash, we try to get the builds to be 100% repeatable, so that we know exactly which version came when.

File: ATParser.lib [hunk 1]
```diff
@@ -1 +1 @@
-https://github.com/ARMmbed/ATParser/#0a74c97ec36659abaa60f25db5f0431c3d054fe7
\ No newline at end of file
+https://github.com/ARMmbed/ATParser/
```
Comment (JanneKiiskila) [inline ATParser.lib]: Fixed hash please for 100% reproducibility on builds.

**Review by JanneKiiskila — CHANGES_REQUESTED** [review]

File: stm-spirit1-rf-driver.lib [hunk 2]
```diff
@@ -1 +1 @@
-https://github.com/ARMmbed/stm-spirit1-rf-driver/#0ff4ca7537f0f2e9137c61ac21251ac06ca42bc3
\ No newline at end of file
+https://github.com/ARMmbed/stm-spirit1-rf-driver/
```
Comment (JanneKiiskila) [inline stm-spirit1-rf-driver.lib]: Same here.

File: wifi-x-nucleo-idw01m1.lib [hunk 3]
```diff
@@ -1 +1 @@
-https://github.com/ARMmbed/wifi-x-nucleo-idw01m1/#465dd7f0fb815fcb0122ef99f4fc69c1519386c5
\ No newline at end of file
+https://github.com/ARMmbed/wifi-x-nucleo-idw01m1/
```
Comment (JanneKiiskila) [inline wifi-x-nucleo-idw01m1.lib]: And here. https://github.com/ARMmbed/easy-connect/pull/46

**Review by JanneKiiskila — CHANGES_REQUESTED** [review]

**Comment by betzw** [comment]
Well, I hoped to be allowed to do so in order to get rid of having to continuously update SHA-1 hashes ... :tired_face: 
Do you think that I can get this _permission_  - at least with respect to my repos?

**Comment by betzw** [comment]
@JanneKiiskila, pls. let me know.

**Comment by JanneKiiskila** [comment]
How often do you think the hashes will really change? Problem being the repeatability of the releases, we do get results on different days if we don't use the hashes. So, for any official stuff - the hashes are pretty much a must, I'm afraid.

**Comment by betzw** [comment]
OK, will update asap!

**Comment by JanneKiiskila** [comment]
Thanks. I've almost done the pull finalization for the [mbed-os-example-client repo](https://github.com/ARMmbed/mbed-os-example-client/tree/X-NUCLEO-IDW01M1), so I could update the hash to match this one as well on the same go.

There's a PR now; https://github.com/ARMmbed/mbed-os-example-client/pull/323 -  @betzw  - your review comments would be greatly appreciated.

**Comment by betzw** [comment]
Updated.

**Review by JanneKiiskila — APPROVED** [review]

**Pull request merged**